### PR TITLE
MAINT: unskip test on win32

### PR DESCRIPTION
**Pull request by mattip**
continuation of gh-15170; following the suggestion of @eric-wieser 

Remove `abspath()`, enable tests on win32

### CODE REVIEW

**Comment by mattip** [comment]
seems to have fixed the problem, the ppc64le test failure is unrelated

File: numpy/random/tests/test_extending.py [hunk 1]
```diff
@@ -40,7 +40,6 @@
 
 @pytest.mark.skipif(cython is None, reason="requires cython")
 @pytest.mark.slow
-@pytest.mark.skipif(sys.platform == 'win32', reason="cmd too long on CI")
 def test_cython(tmp_path):
     examples = os.path.join(os.path.dirname(__file__), '..', '_examples')
     base = os.path.dirname(examples)
```
Comment (eric-wieser) [inline numpy/random/tests/test_extending.py]: Not: this line is dead code

**Review by eric-wieser — APPROVED** [review]

File: numpy/random/tests/test_extending.py (hunk 1 above)
Comment (mattip) [reply numpy/random/tests/test_extending.py]: removed

**Comment by eric-wieser** [comment]
Feel free to merge when CI passes.

**Pull request merged**